applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py

```diff
@@ -0,0 +1,227 @@
+    #mechanical_solution_strategy.InitializeSolutionStep()
+    #mechanical_solution_strategy.Predict()
+    #mechanical_solution_strategy.SolveSolutionStep()
+    #mechanical_solution_strategy.FinalizeSolutionStep()
+
+def AuxiliarCreateConvergenceParameters(main_model_part, settings, contact_settings):
+    # Create an auxiliary Kratos parameters object to store the convergence settings.
+        if contact_settings["fancy_convergence_criterion"].GetBool():
+            table = KM.TableStreamUtility()
+            table.SetOnProcessInfo(main_model_part.ProcessInfo)
+
+        conv_params = KM.Parameters("{}")
+        conv_params.AddValue("convergence_criterion", settings["convergence_criterion"])
+        conv_params.AddValue("rotation_dofs", settings["rotation_dofs"])
+        conv_params.AddValue("echo_level", settings["echo_level"])
+        conv_params.AddValue("displacement_relative_tolerance", settings["displacement_relative_tolerance"])
+        conv_params.AddValue("displacement_absolute_tolerance", settings["displacement_absolute_tolerance"])
+        conv_params.AddValue("residual_relative_tolerance", settings["residual_relative_tolerance"])
+        conv_params.AddValue("residual_absolute_tolerance", settings["residual_absolute_tolerance"])
+        conv_params.AddValue("contact_displacement_relative_tolerance", contact_settings["contact_displacement_relative_tolerance"])
+        conv_params.AddValue("contact_displacement_absolute_tolerance", contact_settings["contact_displacement_absolute_tolerance"])
+        conv_params.AddValue("contact_residual_relative_tolerance", contact_settings["contact_residual_relative_tolerance"])
+        conv_params.AddValue("contact_residual_absolute_tolerance", contact_settings["contact_residual_absolute_tolerance"])
+        conv_params.AddValue("frictional_contact_displacement_relative_tolerance", contact_settings["frictional_contact_displacement_relative_tolerance"])
+        conv_params.AddValue("frictional_contact_displacement_absolute_tolerance", contact_settings["frictional_contact_displacement_absolute_tolerance"])
+        conv_params.AddValue("frictional_contact_residual_relative_tolerance", contact_settings["frictional_contact_residual_relative_tolerance"])
+        conv_params.AddValue("frictional_contact_residual_absolute_tolerance", contact_settings["frictional_contact_residual_absolute_tolerance"])
+        conv_params.AddValue("mortar_type", contact_settings["mortar_type"])
+        conv_params.AddValue("condn_convergence_criterion", contact_settings["condn_convergence_criterion"])
+        conv_params.AddValue("print_convergence_criterion", contact_settings["print_convergence_criterion"])
+        conv_params.AddValue("ensure_contact", contact_settings["ensure_contact"])
+        conv_params.AddValue("frictional_decomposed", contact_settings["frictional_decomposed"])
+        conv_params.AddValue("gidio_debug", contact_settings["gidio_debug"])
```

Comment:
cannot you validate against the other parameters to avoid this manual copying?

Reply:
The structure of the parameters is not the same, I cannot